ako-gateway-api/nodes/avi_model_l7_translator.go

```diff
@@ -75,7 +122,7 @@ func (o *AviObjectGraph) BuildChildVS(key string, routeModel RouteModel, parentN
 		Host:        hosts,
 	}
 	for _, host := range hosts {
-		if !utils.HasElem(parentNode[0].VSVIPRefs[0].FQDNs, host) {
+		if !strings.Contains(host, utils.WILDCARD) && !utils.HasElem(parentNode[0].VSVIPRefs[0].FQDNs, host) {
```

Comment:
But in that case, even if somebody with DNS other than Avi, tries to create the gateway with wildcard in listener hostname, we will still not add that to vsvip. Will the traffic go through for them ?
If not then whole purpose of supporting wildcard in gateway listener hostname is defeated

Reply:
Finding out proper IP address for Given hostname is work of DNS. So even though , implementation is not adding wildcard to vsvip, child vs is capable of handling wildcard requests. So if requests land on Parent VS, then it will be served by appropriate child VS.